netsim/modules/vlan.py

```diff
@@ -100,12 +100,13 @@ def validate_vlan_attributes(obj: Box, topology: Box) -> None:
     if not 'id' in vdata:                                           # When VLAN ID is not defined
       vdata.id = _dataplane.get_next_id('vlan_id')                  # ... take the next free VLAN ID from the list
 
-    vlan_pool = [ vdata.pool ] if isinstance(vdata.get('pool',None),str) else []
-    vlan_pool.extend(['vlan','lan'])
-    pfx_list = links.assign_link_prefix(vdata,vlan_pool,topology.pools,topology.nodes,f'{obj_path}.{vname}')
-    vdata.prefix = addressing.rebuild_prefix(pfx_list)
-    if not 'allocation' in vdata.prefix:
-      vdata.prefix.allocation = 'id_based'
+    if vdata.get('mode','irb')!='route':                            # Don't assign a prefix to routed VLANs
```

Comment:
This might work, but is conceptually wrong.

Either we KNOW the **mode** is set by this time (so there's no need for `vdata.get`, and if that's the case it would be nice to have a comment saying that for our future visits) or we MUST use the standard way to get the VLAN mode (the default might be set in topology).

The way I'm reading the code just above this, the same function sets the VLAN mode only for node VLANs.

Finally, as this is the last thing in the function, it would be nice to use the "left-aligned happy path" approach (= exit if mode = route)

I can make these changes when running the tests tomorrow.

Reply:
> This might work, but is conceptually wrong.

It doesn't even work. The VLAN gets a prefix in the following topology

```
---
provider: clab
defaults.device: frr
module: [ vlan ]

vlan.mode: route
vlans:
  red:

nodes: [ r1,r2 ]
links:
- interfaces: [ r1,r2 ]
  vlan.trunk: [ red ]
```

Reply:
> Either we KNOW the **mode** is set by this time

From looking at the code, ```validate_vlan_attributes``` gets called for the topology first, and then for each node. We refrained from setting vlan.mode on a global vlan because otherwise a node level vlan.mode setting would get ignored

Have updated the code to:
```if vdata.get('mode',default_fwd_mode or 'irb')!='route':```